.github/workflows/ci.yml

```diff
@@ -59,19 +59,35 @@ jobs:
 
       # Run the tests
       - run: just codecov "nextest-unit ${{ matrix.type }}"
-      - run: mv codecov.json unit-${{ matrix.id }}.json
+      - run: mv coverage/codecov/{new,unit-${{matrix.id}}}.json
+      - run: mkdir -p coverage/profraw/{unit,integration,binaries}
+      - run: cd coverage/profraw/new && tar -c --zstd -f ../unit/${{matrix.id}}.tar.zst *
+      - run: rm -rf coverage/profraw/new
```

Comment:
I don't love all this logic inside yaml. In my head a build system rule that produces all of these files as an output would be much more maintainable. In makefile speak:

```
???.tar.zst: unit-tests-or-whatever
    tar -c --zstd ............
```

as it is getting pretty hard to read and also pretty onerous to reproduce locally if one needed so.

Reply:
Hmmmm I think I see your point, but here there are quite a few things that are intertwined, and extracting them all for a build-system production rule would be pretty hard. Not impossible, but would require quite significant changes to the way coverage, at least, is currently implemented. And I’m not sure `make` would be the right tool for the job, though I don’t know of a better tool.

I tried to add a few comments to at least help with the reading cost, is it any better this way? And I agree with you that if/when we switch from `just` to a more build-system-y test runner, we should probably move that all to be in there :)

As for putting it inside `just`, I’m not really sure about it: is it something we expect people to want running manually? As you say, if we were to do it then it’d probably be wobbly and hard-to-use, and I’m not sure anyone except for CI would want to run that locally.

Reply:
My concern is always that there's now more logic to consider in case something is failing on CI but not locally. This was an extremely major pain point for me during the buildkite era and I'm thus severely allergic to any CI-specific logic that does not strictly need to be like that.

For local users it seems to me like it wouldn't be too onerous to add another wrapper rule that does what codecov does today and then this file handling logic as a separate task from plain `codecov` as it is today. That way people looking to reproduce CI behaviour and people looking to generate coverage for local use are both accommodated.

Reply:
Got it, I made this rule in the justfile, it does reduce code duplication a tad :)